tests/sync/app.cpp

```diff
@@ -262,89 +267,141 @@ TEST_CASE("app: UsernamePasswordProviderClient integration", "[sync][app]") {
+            [&](std::shared_ptr<realm::SyncUser> user, Optional<app::AppError> error) {
+                REQUIRE(user);
                 CHECK(!error);
                 processed = true;
-        });
+            });
+        CHECK(processed);
+        processed = false;
+        auto user = app.current_user();
+        REQUIRE(user);
+        CHECK(user->user_profile().email == email);
     }
 
     SECTION("confirm user") {
         app.provider_client<App::UsernamePasswordProviderClient>()
             .confirm_user("a_token",
                           "a_token_id",
                           [&](Optional<app::AppError> error) {
+                REQUIRE(error);
                 CHECK(error->message == "invalid token data");
                 processed = true;
         });
+        CHECK(processed);
     }
 
     SECTION("resend confirmation email") {
         app.provider_client<App::UsernamePasswordProviderClient>()
-            .resend_confirmation_email("username@10gen.com",
+            .resend_confirmation_email(email,
                                        [&](Optional<app::AppError> error) {
+                REQUIRE(error);
                 CHECK(error->message == "already confirmed");
                 processed = true;
         });
+        CHECK(processed);
     }
-    
-    SECTION("send reset password") {
-        app.provider_client<App::UsernamePasswordProviderClient>()
-            .send_reset_password_email("username@10gen.com",
-                                       [&](Optional<app::AppError> error) {
-                CHECK(!error);
-        });
 
-        app.provider_client<App::UsernamePasswordProviderClient>()
-            .send_reset_password_email(email,
-                                       [&](Optional<app::AppError> error) {
-                CHECK(error->message == "user not found");
-                processed = true;
-        });
-    }
-    
-    SECTION("reset password") {
+    SECTION("reset password invalid tokens") {
         app.provider_client<App::UsernamePasswordProviderClient>()
             .reset_password(password,
                             "token_sample",
                             "token_id_sample",
                             [&](Optional<app::AppError> error) {
+                REQUIRE(error);
                 CHECK(error->message == "invalid token data");
                 processed = true;
         });
+        CHECK(processed);
     }
-    
-    SECTION("call reset password function") {
+
+    SECTION("reset password function success") {
+        // the imported test app will accept password reset if the password contains "realm_tests_do_reset" via a function
+        std::string accepted_new_password = util::format("realm_tests_do_reset%1", random_string(10));
         app.provider_client<App::UsernamePasswordProviderClient>()
             .call_reset_password_function(email,
-                                          password,
-                                          "[0,1]",
```

Comment:
This previously failed because this is invalid extended json.

Reply:
^ @ironage if you make the args `"[\"foo\", \"bar\"]`, it will work fine.

Reply:
That works, thanks!